### PR TITLE
Addlivepromos

### LINKED ISSUE #2600: Add LIV to CPS type whitelist
~~Blocked on https://github.com/bbc/psammead/issues/1356~~

**Is your feature request related to a problem? Please describe.**
Similarly to #2559, we need to add 'LIV' to the whitelist array to stop filtering out the live promo assets and integrate the LIV label functionality in Simorgh

**Describe the solution you'd like**
Add LIV to the whitelist array in [cpstypes.js](https://github.com/bbc/simorgh/blob/latest/src/app/lib/utilities/preprocessor/rules/cpstypes.js)

Update story promo container with logic for LIVE label

**Describe alternatives you've considered**
No live promos 😢 

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**
WSFRONTPGS-81

### PR DESCRIPTION
**Pull request by OlgaLyubin**
Resolves #2600 

Requires UX review before merging

**Overall change:** Added lived labels to story promos. Updated the whitelist array to include the 'LIV' cpstype. Added a snapshot test.

**Code changes:**
- Checking if the story promo is live
- Adding a LIVE label to a story promo

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/containers/StoryPromo/index.jsx [hunk 1]
```diff
@@ -81,7 +103,15 @@ const StoryPromo = ({ item, lazyLoadImage, topStory }) => {
       {headline && (
         <Headline script={script} service={service} topStory={topStory}>
           <Link href={url}>
-            <LinkContents item={item} />
+            {isLive ? (
+              <LiveComponent
+                service={service}
+                linkcontents={headline}
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx]: Howcome this is `linkcontents`? Imo `headline` would be better as it doesnt confuse with `linkcontents` which is a JSX node

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/containers/StoryPromo/index.jsx [hunk 2]
```diff
@@ -57,12 +59,30 @@ StoryPromoImage.propTypes = {
   imageValues: shape(storyItem.indexImage).isRequired,
 };
 
+const LiveComponent = ({ linkcontents, service, dir }) => (
```
Comment (DenisHdz) [inline src/app/containers/StoryPromo/index.jsx]:
```suggestion
const LiveComponent = ({ headline, service, dir }) => (
```

File: src/app/containers/StoryPromo/index.jsx [hunk 3]
```diff
@@ -57,12 +59,30 @@ StoryPromoImage.propTypes = {
...
+    </LiveLabel>
+    <VisuallyHiddenText lang="en-GB">Live, </VisuallyHiddenText>
+    {linkcontents}
+  </span>
+);
+
+LiveComponent.propTypes = {
+  service: string.isRequired,
+  dir: string.isRequired,
+  linkcontents: element.isRequired,
```
Comment (DenisHdz) [inline src/app/containers/StoryPromo/index.jsx]:
```suggestion
  headline: element.isRequired,
```

File: src/app/containers/StoryPromo/index.jsx [hunk 4]
```diff
@@ -82,7 +104,15 @@ const StoryPromo = ({ item, lazyLoadImage, topStory }) => {
       {headline && (
         <Headline script={script} service={service} topStory={topStory}>
           <Link href={url}>
-            <LinkContents item={item} />
+            {isLive ? (
+              <LiveComponent
+                service={service}
+                linkcontents={linkcontents}
```
Comment (DenisHdz) [inline src/app/containers/StoryPromo/index.jsx]:
```suggestion
                headline={linkcontents}
```

**Review by DenisHdz — COMMENTED** [review]
Great job on your first PR 🙂
Just to clarify, I think that @dr3 meant to call the prop `headline`, as we have in the `Psammead` story.

**Review by j-pendlebury — CHANGES_REQUESTED** [review]
What @DenisBBC said. You can use Github's UI to commit the changes by clicking on "Commit Suggestion" :)

**Review by DenisHdz — APPROVED** [review]
LGTM :)

File: src/app/containers/StoryPromo/index.jsx [hunk 5]
```diff
@@ -58,12 +60,30 @@ StoryPromoImage.propTypes = {
   imageValues: shape(storyItem.indexImage).isRequired,
 };
 
+const LiveComponent = ({ headline, service, dir }) => (
+  // eslint-disable-next-line jsx-a11y/aria-role
+  <span role="text">
+    <LiveLabel service={service} dir={dir}>
+      LIVE
+    </LiveLabel>
+    <VisuallyHiddenText lang="en-GB">Live, </VisuallyHiddenText>
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx:79]: Is there a translation for this? Is there an issue to add it?

File: src/app/containers/StoryPromo/index.jsx [hunk 6]
```diff
@@ -82,7 +104,15 @@ const StoryPromo = ({ item, lazyLoadImage, topStory }) => {
       {headline && (
         <Headline script={script} service={service} topStory={topStory}>
           <Link href={url}>
-            <LinkContents item={item} />
+            {isLive ? (
+              <LiveComponent
+                service={service}
+                headline={linkcontents}
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx:112]: Is this right? I meant `headline={headline}`

Comment (DenisHdz) [reply src/app/containers/StoryPromo/index.jsx:112]: `const linkcontents = <LinkContents item={item} />; `

File: src/app/containers/StoryPromo/index.jsx (hunk 5 above)
Comment (OlgaLyubin) [reply src/app/containers/StoryPromo/index.jsx:79]: This is a deliberate design choice not to translate the LIVE label.

Comment (dr3) [reply src/app/containers/StoryPromo/index.jsx:79]: Awesome :)

File: src/app/containers/StoryPromo/index.jsx (hunk 6 above)
Comment (dr3) [reply src/app/containers/StoryPromo/index.jsx]: Chatted IRL, using `linkcontents`, not `headline` is to make use of `overtyped || headline` in `linkcontents` :)

**Review by dr3 — APPROVED** [review]
Chatted IRL, using `linkcontents`, not `headline` is to make use of `overtyped || headline` in `linkcontents` :)

LGTM, nice work

**Review by j-pendlebury — APPROVED** [review]
Yay! Congrats on your first PR :)

**Comment by david-boydell** [comment]
All good, this is ready for merge.

File: src/app/containers/StoryPromo/index.jsx [hunk 7]
```diff
@@ -68,13 +70,33 @@ StoryPromoImage.defaultProps = {
   }),
 };
 
+const LiveComponent = ({ headline, service, dir }) => (
+  // eslint-disable-next-line jsx-a11y/aria-role
+  <span role="text">
+    <LiveLabel service={service} dir={dir}>
```
Comment (greenc05) [inline src/app/containers/StoryPromo/index.jsx:76]: Does this have an aria-hidden attribute @OlgaLyubin

Comment (greenc05) [reply src/app/containers/StoryPromo/index.jsx:76]: And also just checking if this has had a developer accessibility swarm carried out on it?

Comment (andrew-nowak) [reply src/app/containers/StoryPromo/index.jsx:76]: Yes this has an aria-hidden attribute - see https://github.com/bbc/psammead/blob/038a0bf5a4f2efd9792eaec0becec79cfbd56ad7/packages/components/psammead-story-promo/src/index.jsx#L260
Yes this has had a swarm - see https://github.com/bbc/psammead/pull/1359#issuecomment-516376021

Comment (greenc05) [reply src/app/containers/StoryPromo/index.jsx:76]: Fabulous, thank you @aacn500

**Pull request merged**